django/conf/urls/__init__.py

```diff
@@ -86,28 +88,35 @@ def patterns(prefix, *args):
     pattern_list = []
     for t in args:
         if isinstance(t, (list, tuple)):
-            t = url(prefix=prefix, *t)
-        elif isinstance(t, RegexURLPattern):
-            t.add_prefix(prefix)
+            if len(t) != 2 or not isinstance(t[1], BaseResolver):
+                t = url(prefix=prefix, *t)
+            elif len(t) == 2 and isinstance(t[1], ResolverEndpoint):
+                t[1].add_prefix(prefix)
         pattern_list.append(t)
     return pattern_list
 
 
-def url(regex, view, kwargs=None, name=None, prefix=''):
+def url(constraints, view, kwargs=None, name=None, prefix='', decorators=None):
+    if isinstance(constraints, six.string_types):
+        constraints = RegexPattern(constraints)
+    elif isinstance(constraints, Promise):
+        constraints = LocalizedRegexPattern(constraints)
+    if not isinstance(constraints, (list, tuple)):
```

Comment:
The previous `if/elif` converts `constraints` into either a `RegexPattern` or `LocalizedRegexPattern` but then immediately has to re-convert it into a list containing one of said objects. Would it make sense to have lines 100 & 103 both be wrapped into `[]`?